prime-router/src/main/kotlin/azure/HistoryFunctions.kt

```diff
@@ -290,40 +296,47 @@ open class BaseHistoryFunction : Logging {
                 OffsetDateTime.now().minusDays(DAYS_TO_SHOW),
                 organizationName ?: authClaims.organization.name
             )
-
             @Suppress("NEW_INFERENCE_NO_INFORMATION_FOR_PARAMETER")
-            val reports = headers.sortedByDescending { it.createdAt }.map {
-                var facilities = arrayListOf<Facility>()
-                if (it.bodyFormat == "CSV")
-                    try {
-                        facilities = getFieldSummaryForReportId(
-                            arrayOf("Testing_lab_name", "Testing_lab_CLIA"), it.reportId.toString(), authClaims
-                        )
-                    } catch (ex: Exception) {
-                        // context.logger.info( "Exception during getFieldSummaryForReportId - TestingLabName was not found - no facilities data will be published" );
-                    }
-                val actions = getActionsForReportId(it.reportId.toString(), authClaims)
-
-                val header = workflowEngine.fetchHeader(it.reportId, authClaims.organization)
-
-                val content = if (header.content !== null) String(header.content) else ""
-                val filename = Report.formExternalFilename(header)
-                val mimeType = Report.Format.safeValueOf(header.reportFile.bodyFormat).mimeType
+            val reports = headers.sortedByDescending { it.createdAt }.mapNotNull {
+                val facilities = arrayListOf<Facility>()
+                val actions = arrayListOf<Action>()
+                // get the org passed in
+                val adminOrg = workflowEngine.settings.organizations.firstOrNull { org ->
+                    org.name.lowercase() == organizationName
+                }
+                val header = try {
+                    workflowEngine.fetchHeader(it.reportId, adminOrg ?: authClaims.organization)
+                } catch (ex: Exception) {
+                    context.logger.severe(ex.message)
+                    context.logger.severe(ex.stackTraceToString())
+                    null
+                }
 
-                ReportView.Builder()
-                    .reportId(it.reportId.toString())
-                    .sent(it.createdAt.toEpochSecond() * 1000)
-                    .via(it.bodyFormat)
-                    .total(it.itemCount.toLong())
-                    .fileType(it.bodyFormat)
-                    .type("ELR")
-                    .expires(it.createdAt.plusDays(DAYS_TO_SHOW).toEpochSecond() * 1000)
-                    .facilities(facilities)
-                    .actions(actions)
-                    .content(content)
-                    .fileName(filename)
-                    .mimeType(mimeType)
-                    .build()
+                if (header != null) {
+                    val content = if (header.content !== null) String(header.content) else ""
```

Comment:
Seems like 
    `val content = header.content ?: ""`
might be sufficient here?

Also note the use of the !== is a pointer/reference check, which is probably harmless when looking for nulls, but generally may not be what you mean here.